dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -210,28 +211,31 @@ public void close() {
 
         @Override
         public T deserialize(Response response) {
-            try {
-                if (errorDecoder.isError(response)) {
+            if (errorDecoder.isError(response)) {
```

Comment:
we may want to use a defensive pattern here to handle unexpected cases:

```java
boolean closeResponse = true;
try {
  /* work*/
  T result = deserializer.deserialize(response.body(), response);
   closeResponse = false; // deserialier is responsible for closing resources when necessary
   return result;
} finally {
    if (closeResponse) response.close();
}
```

Reply:
Oh yeah that's much nicer - will update